src/Promitor.Core.Scraping/Scraper.cs

```diff
@@ -84,9 +84,45 @@ public async Task ScrapeAsync(MetricDefinition metricDefinition)
 
                 var metricsTimestampFeatureFlag = FeatureFlag.IsActive(FeatureFlag.Names.MetricsTimestamp, defaultFlagState: true);
 
-                var gauge = Metrics.CreateGauge(metricDefinition.Name, metricDefinition.Description, includeTimestamp: metricsTimestampFeatureFlag, labelNames:"resource_uri");
+                var gauge = Metrics.CreateGauge(metricDefinition.Name, metricDefinition.Description, includeTimestamp: metricsTimestampFeatureFlag, labelNames: "resource_uri");
                 gauge.WithLabels(scrapedMetricResult.ResourceUri).Set(scrapedMetricResult.MetricValue);
             }
+            catch (ErrorResponseException erex)
+            {
+                string reason = string.Empty;
+
+                if (!string.IsNullOrEmpty(erex.Message))
+                {
+                    reason = erex.Message;
+                }
+
+                if (erex.Response != null && !string.IsNullOrEmpty(erex.Response.Content))
+                {
+                    try
+                    {
+                        var definition = new { error = new { code = "", message = "" } };
+                        var jsonError = JsonConvert.DeserializeAnonymousType(erex.Response.Content, definition);
+
+                        if (jsonError != null && jsonError.error != null)
+                        {
+                            if (!string.IsNullOrEmpty(jsonError.error.message))
+                            {
+                                reason = $"{jsonError.error.code}: {jsonError.error.message}";
+                            }
+                            else if (!string.IsNullOrEmpty(jsonError.error.code))
+                            {
+                                reason = $"{jsonError.error.code}";
+                            }
+                        }
+                    }
+                    catch (Exception)
+                    {
+                        // do nothing. maybe a bad deserialization of json content. Just fallback on outer exception message.

```

Comment:
If it fails, let's just track the original exception

```suggestion
                        // do nothing. maybe a bad deserialization of json content. Just fallback on outer exception message.
                        _exceptionTracker.Track(errorResponseException);

```